[Ingest pipelines] Address copy feedback

alisonelizabeth opened the pull request:
This PR updates copy in the Ingest Node Pipelines app.

### Screenshots

<img width="1253" alt="Screen Shot 2020-05-04 at 2 17 14 PM" src="https://user-images.githubusercontent.com/5226211/81027547-38cf7480-8e4c-11ea-8535-3eb7c0f8eaee.png">

<img width="1232" alt="Screen Shot 2020-05-04 at 4 17 59 PM" src="https://user-images.githubusercontent.com/5226211/81027554-3ff68280-8e4c-11ea-8939-c725fb478b6a.png">

![create](https://user-images.githubusercontent.com/5226211/81028159-4dad0780-8e4e-11ea-9d17-83644d2dfed8.png)

<img width="563" alt="Screen Shot 2020-05-04 at 9 26 51 PM" src="https://user-images.githubusercontent.com/5226211/81028171-569dd900-8e4e-11ea-8556-9a6fe1e91492.png">

<img width="546" alt="Screen Shot 2020-05-04 at 9 28 54 PM" src="https://user-images.githubusercontent.com/5226211/81028173-57366f80-8e4e-11ea-97e9-58dbc25db5dd.png">

<img width="1251" alt="Screen Shot 2020-05-04 at 9 30 42 PM" src="https://user-images.githubusercontent.com/5226211/81028241-8a78fe80-8e4e-11ea-9e75-2affa6fa304d.png">

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/)
* Commit: 5ef63d8c544b92bbf6943e40807921f0b4d2a558

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/","number":"45254","commit":"5ef63d8c544b92bbf6943e40807921f0b4d2a558"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_test_flyout/tabs/tab_output.tsx:
```diff
@@ -59,7 +59,7 @@ export const OutputTab: React.FunctionComponent<Props> = ({
           <FormattedMessage
             id="xpack.ingestPipelines.testPipelineFlyout.outputTab.descriptionText"
-            defaultMessage="The output of the executed pipeline. {runLink}"
+            defaultMessage="View the final output data, or see how each processor affects the ingest document as it passes through the pipeline. {runLink}"
```
@gchaps Let me know what you think of this (screenshot in initial PR comment). This differs from your original suggestion, as the `View verbose output` is actually disabled by default.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/)
* Commit: 8454ef220474325524c2e80abd32e7834a855a73
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/execution/node/261/log)

### History
* :green_heart: [Build #45254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/) succeeded 5ef63d8c544b92bbf6943e40807921f0b4d2a558

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/","number":"45254","commit":"5ef63d8c544b92bbf6943e40807921f0b4d2a558"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/","number":"45359","commit":"8454ef220474325524c2e80abd32e7834a855a73"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45458/)
* Commit: 0511a3cd682890b3a68752c8d875e42af936b7cb

### History
* :broken_heart: [Build #45359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/) failed 8454ef220474325524c2e80abd32e7834a855a73
* :green_heart: [Build #45254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/) succeeded 5ef63d8c544b92bbf6943e40807921f0b4d2a558

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/","number":"45254","commit":"5ef63d8c544b92bbf6943e40807921f0b4d2a558"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/","number":"45359","commit":"8454ef220474325524c2e80abd32e7834a855a73"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45458/","number":"45458","commit":"0511a3cd682890b3a68752c8d875e42af936b7cb"}]}
PIPELINE-->

cjcenizal commented on x-pack/plugins/ingest_pipelines/public/application/app.tsx:
```diff
@@ -71,16 +71,32 @@ export const App: FunctionComponent = () => {
+                          return (
+                            <>
+                              {acc}{' '}
+                              <FormattedMessage
```
I think it's worth checking with the i18n folks, but I don't think that we can depend on all languages following the English structure of using a word ("and" in the case of English) to join the last item onto a list. It might be safer to use the original pattern of simply joining on a comma.

alisonelizabeth replied:
That's a good point. I reverted the change to the original implementation. This also follows the pattern we're using in Snapshot + Restore, so it will be consistent.

cjcenizal commented on x-pack/plugins/ingest_pipelines/public/application/sections/pipelines_list/main.tsx:
```diff
@@ -179,9 +179,22 @@ export const PipelinesList: React.FunctionComponent<RouteComponentProps> = ({
+              title={
+                <FormattedMessage
+                  id="xpack.ingestPipelines.list.loadErrorTitle"
+                  defaultMessage="Unable to load pipelines. Try {reloadLink} the page."
```
The way this text is phrased, it seems like clicking the reload link will reload the page, which I think of as a browser page refresh. But clicking this link will actually just resend the request to load the pipelines. Can we align text with action more closely by phrasing this as "Try reloading them"?

alisonelizabeth replied:
Good point! I went with Gail's suggestion: "Try again".

gchaps commented:
I'm going to rescind one of my earlier comments.  I think the text reads better as  "**Provide** an array of documents for the pipeline to ingest."

Per @cjcenizal, how about:

Unable to load pipelines.  Try again.

and make "Try again" the link.

The text in the Output screen felt a little too long for me.  How about we go back to using an intro and tooltip:

Intro:  View the final output data.
Tooltip: See how each processor affects the document as it passes through the pipeline.

cjcenizal commented:
> Unable to load pipelines. Try again.

RGTM!

> The text in the Output screen felt a little too long for me. How about we go back to using an intro and tooltip:

I'm not sure the intro/tooltip option is equivalent with what we currently have. The current text offers the user two exclusive options: final output data _or_ data per-processor. But your suggestion makes it sound like the tooltip text is an elaboration upon the intro text.

cchaos commented:
Hey @alisonelizabeth I just glanced at your screenshots and have one request. Can you separate create a single empty state panel instead of putting the empty state underneath the title for the listing page? It just creates a weird hierarchy as it is and then you can explain more of what a pipeline is and duplicate the docs link inside of it.

<img width="1251" alt="81028241-8a78fe80-8e4e-11ea-9e75-2affa6fa304d" src="https://user-images.githubusercontent.com/549577/81093069-8e993080-8ecf-11ea-815f-d39639a0be9f.png">

alisonelizabeth commented:
@cchaos thanks for the catch! I will open up a separate PR and address this.

gchaps commented:
**Empty state**

How about using "Learn more" followed by an icon to let users know they are going to an outside doc.

**Output**

> I'm not sure the intro/tooltip option is equivalent with what we currently have. 

We could keep the intro (I removed a few words) and then use a space to separate the intro from the Refresh link so it doesn't feel so crowded)

View the output data, or see how each processor affects the document as it passes through the pipeline.

**Failure processors**

What about adding the word "alternate".  Some choices:

- The alternate processors to execute after a processor fails.
- The alternate processors to execute after the pipeline stops because of a failure.
- The alternate processors to execute after an original processor in the pipeline fails.

alisonelizabeth commented:
@gchaps @cjcenizal I believe I addressed all of your feedback. Thanks for the review! Updated screenshots:

Empty prompt
<img width="1259" alt="Screen Shot 2020-05-05 at 3 56 41 PM" src="https://user-images.githubusercontent.com/5226211/81111208-b2b63b00-8eea-11ea-9302-87d29b1efa98.png">

Output tab - I'm not sure if I'm sold on the "Refresh output" link on its own line. @cchaos WDYT? 
<img width="558" alt="Screen Shot 2020-05-05 at 4 01 54 PM" src="https://user-images.githubusercontent.com/5226211/81111209-b2b63b00-8eea-11ea-8edc-fc53d778a630.png">

Create form
<img width="1258" alt="Screen Shot 2020-05-05 at 4 03 06 PM" src="https://user-images.githubusercontent.com/5226211/81111211-b34ed180-8eea-11ea-8a90-7e6a3fc2d4d8.png">

Error
<img width="1264" alt="Screen Shot 2020-05-05 at 12 45 45 PM" src="https://user-images.githubusercontent.com/5226211/81111213-b34ed180-8eea-11ea-96c4-9b4b018fb4d6.png">

cchaos commented:
Thanks @alisonelizabeth for going ahead and tackling the empty screen! Could you also add a small line break before the "Learn more" link? Just to ensure that there isn't a line break inside of the link itself.

<img width="356" alt="Image 2020-05-05 at 5 31 20 PM" src="https://user-images.githubusercontent.com/549577/81118300-a46e1c00-8ef6-11ea-8316-f449412d7b0e.png">


As for the refresh button. AFAICT, It refreshes the code block below? If it's crammed up with the description text above then it seems more like a link than a button. I would even go further and maybe make it a proper button (the borderd, but small kind). 

You could also put the refresh button and the toggle in the same line with the toggle on the far right. So then it looks like a toolbar for the code output.

<img width="547" alt="Image 2020-05-05 at 5 32 06 PM" src="https://user-images.githubusercontent.com/549577/81118490-f44ce300-8ef6-11ea-92bf-88cd15bf3cd7.png">

gchaps commented:
Text looks good.  I have one small comment.  For the error message, can we use "Try again"?

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45701/)
* Commit: fbcfb7cb2a793d70438b8710a03fbfaf8c0e1666
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45701/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45701/execution/node/208/log)

### History
* :green_heart: [Build #45458](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45458/) succeeded 0511a3cd682890b3a68752c8d875e42af936b7cb
* :broken_heart: [Build #45359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/) failed 8454ef220474325524c2e80abd32e7834a855a73
* :green_heart: [Build #45254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/) succeeded 5ef63d8c544b92bbf6943e40807921f0b4d2a558

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/","number":"45254","commit":"5ef63d8c544b92bbf6943e40807921f0b4d2a558"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/","number":"45359","commit":"8454ef220474325524c2e80abd32e7834a855a73"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45458/","number":"45458","commit":"0511a3cd682890b3a68752c8d875e42af936b7cb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45701/","number":"45701","commit":"fbcfb7cb2a793d70438b8710a03fbfaf8c0e1666"}]}
PIPELINE-->

alisonelizabeth commented:
> Text looks good. I have one small comment. For the error message, can we use "Try again"?

@gchaps oops! The screenshot I added was outdated. The text is "Try again" now.

alisonelizabeth commented:
Thanks @cchaos! I added a line break to the empty prompt and also applied your suggestion regarding the "Reload" functionality. I think this looks a lot better - thanks! 

<img width="576" alt="Screen Shot 2020-05-05 at 9 05 33 PM" src="https://user-images.githubusercontent.com/5226211/81130227-fb82e980-8f14-11ea-8a88-cb5e6a0bf52b.png">

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45809/)
* Commit: 569a4f424c77a980669037a550d66eefdc1a33d7

### History
* :broken_heart: [Build #45701](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45701/) failed fbcfb7cb2a793d70438b8710a03fbfaf8c0e1666
* :green_heart: [Build #45458](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45458/) succeeded 0511a3cd682890b3a68752c8d875e42af936b7cb
* :broken_heart: [Build #45359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/) failed 8454ef220474325524c2e80abd32e7834a855a73
* :green_heart: [Build #45254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/) succeeded 5ef63d8c544b92bbf6943e40807921f0b4d2a558

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45254/","number":"45254","commit":"5ef63d8c544b92bbf6943e40807921f0b4d2a558"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45359/","number":"45359","commit":"8454ef220474325524c2e80abd32e7834a855a73"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45458/","number":"45458","commit":"0511a3cd682890b3a68752c8d875e42af936b7cb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45701/","number":"45701","commit":"fbcfb7cb2a793d70438b8710a03fbfaf8c0e1666"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45809/","number":"45809","commit":"569a4f424c77a980669037a550d66eefdc1a33d7"}]}
PIPELINE-->

cjcenizal commented:
I’m going to merge this so we make FF.
